### PR TITLE
DOC: update scipy.integrate section

### PR DESCRIPTION
**Pull request by mdhaber**

### CODE REVIEW

File: intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py [hunk 1]
```diff
@@ -10,24 +10,24 @@
 import scipy as sp
 import matplotlib.pyplot as plt
 
-mass = 0.5  # kg
```
Comment (mdhaber) [inline intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py:13]: Rename to facilitate comparison of code and math.

File: intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py [hunk 2]
```diff
@@ -10,24 +10,24 @@
...
+t_span = (0, 10)
+t_eval = np.linspace(*t_span, 100)
+z0 = [1, 0]
+res = sp.integrate.solve_ivp(f, t_span, z0, args=(zeta, omega),
+                             method = 'LSODA')
 
 plt.figure(figsize=(4, 3))
-plt.plot(solution.t, solution.y[0,:], label='y')
-plt.plot(solution.t, solution.y[1,:], label="y'")
+plt.plot(res.t, res.y[0], label='y')
```
Comment (mdhaber) [inline intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py:30]: Use standard `res` for all result objects.

File: intro/scipy/examples/plot_solve_ivp_simple.py [hunk 3]
```diff
@@ -11,14 +11,18 @@
...
 plt.figure(figsize=(4, 3))
-plt.plot(solution.t, solution.y[0,:])
-plt.xlabel('t: Time')
-plt.ylabel('y: Position')
+plt.plot(res.t, res.y[0])
+plt.xlabel('t')
+plt.ylabel('y')
+plt.title('Solution of Initial Value Problem')
 plt.tight_layout()
+plt.show()
```
Comment (mdhaber) [inline intro/scipy/examples/plot_solve_ivp_simple.py:28]:
```suggestion
plt.show()

```

File: intro/scipy/index.rst [hunk 4]
```diff
@@ -751,88 +751,116 @@ whether the means of two sets of observations are significantly different::
...
 .. image:: auto_examples/images/sphx_glr_plot_solve_ivp_simple_001.png
     :target: auto_examples/plot_solve_ivp_simple.html
     :scale: 70
     :align: right
 
+Let us integrate a more complex ODE: a `damped
+spring-mass oscillator
+<https://en.wikipedia.org/wiki/Harmonic_oscillator#Damped_harmonic_oscillator>`__.
+The position of a mass attached to a spring obeys the 2nd order ODE
+:math:`\ddot{y} + 2 \zeta \omega_0  \dot{y} + \omega_0^2 y = 0` with natural frequency
```
Comment (mdhaber) [inline intro/scipy/index.rst:818]: Use of $\zeta$ for the [damping ratio](https://en.wikipedia.org/wiki/Damping) is nearly universal; I've not seen $\varepsilon$ used aside from here.

File: intro/scipy/index.rst [hunk 5]
```diff
@@ -751,88 +751,116 @@ whether the means of two sets of observations are significantly different::
...
 
-    >>> eps < 1
-    True
+    >>> m = 0.5  # kg
+    >>> k = 4  # N/m
+    >>> c = 0.4  # N s/m
+    >>> zeta = c / (2 * m * np.sqrt(k/m))
+    >>> omega = np.sqrt(k / m)
 
-For :func:`~scipy.integrate.solve_ivp`, the 2nd order equation
```
Comment (mdhaber) [inline intro/scipy/index.rst:829]: This is often tricky for those who are unfamiliar with the process. I tried to make it more explicit.

File: intro/scipy/index.rst [hunk 6]
```diff
@@ -841,20 +869,17 @@ velocity and acceleration::
...
-
-.. raw:: html
-
-   <div style="clear: both"></div>
-
+    >>> t_span = (0, 10)
+    >>> t_eval = np.linspace(*t_span, 100)
+    >>> z0 = [1, 0]
+    >>> res = sp.integrate.solve_ivp(f, t_span, z0, args=(zeta, omega),
+    ...                              method='LSODA')
```
Comment (mdhaber) [inline intro/scipy/index.rst:876]:
```suggestion
    >>> res = sp.integrate.solve_ivp(f, t_span, z0, t_eval=t_eval,
    ...                              args=(zeta, omega), method='LSODA')
```

File: intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py [hunk 7]
```diff
@@ -10,24 +10,24 @@
...
-    return (yvec[1], -eps * omega * yvec[1] - omega **2 * yvec[0])
-
-time_span = (0, 10)
-yinit = (1, 0)
-solution = sp.integrate.solve_ivp(calc_deri, time_span, yinit, args=(eps, omega), method='LSODA')
+t_span = (0, 10)
+t_eval = np.linspace(*t_span, 100)
+z0 = [1, 0]
+res = sp.integrate.solve_ivp(f, t_span, z0, args=(zeta, omega),
+                             method = 'LSODA')
```
Comment (mdhaber) [inline intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py:27]:
```suggestion
res = sp.integrate.solve_ivp(f, t_span, z0, t_eval=t_eval,
                             args=(zeta, omega), method = 'LSODA')
```

File: intro/scipy/index.rst [hunk 8]
```diff
@@ -751,88 +751,116 @@ whether the means of two sets of observations are significantly different::
...
-.. note::
+    \frac{dy}{dt} = f(t, y(t))
+
+from an initial time :math:`t_0` and initial state :math:`y(t=t_0)=t_0` to a final
+time :math:`t_f` or until an event occurs (e.g. a specified state is reached).
+
+As an introduction, consider the initial value problem given by
+:math:`\frac{dy}{dt} = -2 y` and the initial condition :math:`y(t=0) = 1` on
+the interval :math:`t = 0 \dots 4`. We begin by defining a callable that
+computes the :math:`f(t, y(t))` given the current time and state.
```
Comment (mdhaber) [inline intro/scipy/index.rst:790]:
```suggestion
computes :math:`f(t, y(t))` given the current time and state.
```

File: intro/scipy/index.rst [hunk 9]
```diff
@@ -751,88 +751,116 @@ whether the means of two sets of observations are significantly different::
...
 
-Function integrals
-...................
+Quadrature
+..........
 
-The most generic integration routine is :func:`scipy.integrate.quad`. To
-compute :math:`\int_0^{\pi / 2} sin(t) dt`::
+Suppose we wish to compute the definite integral
+:math:`\int_0^{\pi / 2} sin(t) dt` numerically. :func:`scipy.integrate.quad`
```
Comment (mdhaber) [inline intro/scipy/index.rst:758]:
```suggestion
:math:`\int_0^{\pi / 2} \sin(t) dt` numerically. :func:`scipy.integrate.quad`
```

File: intro/scipy/index.rst [hunk 10]
```diff
@@ -751,88 +751,116 @@ whether the means of two sets of observations are significantly different::
...
+chooses one of several adaptive techniques depending on the parameters, and
+therefore is a reasonable first choice for integration of a single variable::
```
Comment (mdhaber) [inline intro/scipy/index.rst:760]:
```suggestion
is therefore the recommended first choice for integration of a single variable::
```

File: intro/scipy/index.rst [hunk 11]
```diff
@@ -751,88 +751,116 @@ whether the means of two sets of observations are significantly different::
...
-For :func:`~scipy.integrate.solve_ivp`, the 2nd order equation
-needs to be transformed in a system of two first-order equations for the
-vector :math:`Y = (y, y')`: the function computes the
-velocity and acceleration::
+and define the function that computes :math:`\dot{z} = f(t, z(t))`::
 
-    >>> def calc_deri(time, yvec, eps, omega):
-    ...     return (yvec[1], -2.0 * eps * omega * yvec[1] - omega **2 * yvec[0])
+    >>> def f(t, z, zeta, omega):
+    ...     return (z[1], -2.0 * zeta * omega * z[1] - omega **2 * z[0])
```
Comment (mdhaber) [inline intro/scipy/index.rst:863]:
```suggestion
    ...     return (z[1], -2.0 * zeta * omega * z[1] - omega**2 * z[0])
```

File: intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py [hunk 12]
```diff
@@ -10,24 +10,24 @@
...
+k = 4  # N/m
+c = 0.4  # N s/m
 
+zeta = c / (2 * m * np.sqrt(k/m))
+omega = np.sqrt(k / m)
 
-eps = cviscous / (2 * mass * np.sqrt(kspring/mass))
-omega = np.sqrt(kspring / mass)
+def f(t, z, zeta, omega):
+    return (z[1], -zeta * omega * z[1] - omega **2 * z[0])
```
Comment (mdhaber) [inline intro/scipy/examples/plot_solve_ivp_damped_spring_mass.py:21]:
```suggestion
    return (z[1], -zeta * omega * z[1] - omega**2 * z[0])
```

**Comment by mdhaber** [comment]
I'm not sure how to fix the issues with use of the `.. math::` directive.

**Review by jarrodmillman — APPROVED** [review]

**Comment by mdhaber** [comment]
I should have marked this as draft. It's not rendering correctly.

**Pull request merged**